packages/app/src/cli/services/deploy.ts

```diff
@@ -82,9 +82,16 @@ export async function deploy(options: DeployOptions) {
           title: unifiedDeployment ? 'Creating deployment' : 'Pushing your code to Shopify',
           task: async () => {
             const extensions = await Promise.all(
-              options.app.allExtensions.flatMap((ext) =>
-                ext.bundleConfig({identifiers, token, apiKey, unifiedDeployment}),
-              ),
+              options.app.allExtensions.flatMap(async (ext) => {
+                const bundleConfig = await ext.bundleConfig({identifiers, token, apiKey, unifiedDeployment})
+                return (
+                  bundleConfig && {
+                    uuid: bundleConfig.uuid,
+                    context: bundleConfig.context,
+                    config: JSON.stringify(bundleConfig.config),
```

Comment:
With the change I proposed in `extension-instance` this shouldn't be necessary 👌